### ISSUE TITLE
Users unable to edit certificate node 

### ISSUE THREAD

**Issue by jynxiesue** [knieper/crltmich#6]
Owner should be able to edit the node when the node status is "Needs Work (revision requested)".
The only editable field should be "Comments".
Once the edit is finished and saved, the workflow state should change to "New Request (waiting for review)" for A & B or "Needs Verification" for C1, C2, and D .
Once status changed, owner cannot edit anymore.

**Comment by jynxiesue** [comment]
The above functionality used to work but got broken sometime over the summer possibly when grants was deployed.

**Comment by knieper** [comment]
I noticed in the code that there is logic that says when creating a new certificate if it is B2, C1, C2, or D it goes directly to Needs verification. This sounds slightly different from the desired logic described here for editing certificates so I want to make sure that it is correct. I do not believe this code was running before based on an error. Plan to review on Monday 2-29-16 call.

**Comment by knieper** [comment]
Based on 2-29-16 phone call B2 certs upon creation do NOT go to needs verification but goes to the queue.

**Comment by jynxiesue** [comment]
Owner can now edit the comments, but I made a mistake, the owner should also be able to edit the semester.  Please enable on edit the same fields as on add mode, which differs depending on certificate type.
In testing, the administrator changed the status (to "Needs Work"), but the owner never got an email message notifying them that the status of the record had been changed.  
When owner creates and saves a new certificate, they get a blank white screen, however, the certificate has been created.
When the owner edits and saves a certificate, they get a "Server Error 500" message, however, the edit is saved. Sometimes the edits are saved twice with the same stamp.
All of these have been replicated across at least two browsers.

**Comment by knieper** [comment]
Currently on staging site the 'demo' setting is set meaning all notification emails will be sent to me.

**Comment by mbakewel** [comment]
We did get a white screen sometimes when creating certificates - the node was created bu instead of getting the node view after, we got a white screen some times but not always. Otherwise this seems to be working well now!

**Comment by mbakewel** [comment]
We retested all cert types and all workflow transitions and edit permissions worked properly. We did encounter white screens and 500 errors several times but not consistently. Screen shots show time stamp and url that we were trying to access. 
![image](https://cloud.githubusercontent.com/assets/5739661/14211913/85f86124-f7fe-11e5-96b6-e2b9d8b9f79e.png)
![image](https://cloud.githubusercontent.com/assets/5739661/14211930/8ec3d176-f7fe-11e5-9cac-a9559356ed7c.png)
